src/Gui/CommandPyImp.cpp

```diff
@@ -322,16 +325,18 @@ PyObject* CommandPy::getAction(PyObject *args)
 }
 
 
-PyObject* CommandPy::createCustomCommand(PyObject* args)
+PyObject* CommandPy::createCustomCommand(PyObject* args, PyObject* kw)
 {
-    const char* macroFile = nullptr;
+    const char* macroFile;
     const char* menuTxt = nullptr;
     const char* tooltipTxt = nullptr;
     const char* whatsthisTxt = nullptr;
     const char* statustipTxt = nullptr;
     const char* pixmapTxt = nullptr;
     const char* shortcutTxt = nullptr;
-    if (!PyArg_ParseTuple(args, "s|zzzzzz", &macroFile, &menuTxt, &tooltipTxt, &whatsthisTxt, &statustipTxt, &pixmapTxt, &shortcutTxt))
```

Comment:
It's important that it's explicitly allowed to pass None for the optional arguments because it's already used like that in the AddonManager. I modified your PR accordingly.

Reply:
Ok!